### PR TITLE
Deprecate dynamic log sanitization

### PR DESCRIPTION
**Pull request by ehashman**
#### What type of PR is this?

/kind deprecation

#### What this PR does / why we need it:

Removes dynamic log sanitization feature.

#### Which issue(s) this PR fixes:

Addresses https://github.com/kubernetes/enhancements/issues/1753

#### Special notes for your reviewer:

See https://github.com/kubernetes/enhancements/pull/3096

#### Does this PR introduce a user-facing change?

```release-note
ACTION REQUIRED: The experimental dynamic log sanitization feature has been deprecated and removed in the 1.24 release. The feature is no longer available for use.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: http://kep.k8s.io/1753
```

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -1937,12 +1937,10 @@ k8s.io/component-base/configz
 k8s.io/component-base/featuregate
 k8s.io/component-base/featuregate/testing
 k8s.io/component-base/logs
-k8s.io/component-base/logs/datapol
```
Comment (ehashman) [inline vendor/modules.txt:1940]: Note: we still need the [data policy tags for static analysis](https://github.com/kubernetes/kubernetes/blob/1d52f442ee745f38a190e0b0563405b6b30a3750/hack/testdata/levee/levee-config.yaml#L25-L32) so I left them alone, but I believe this isn't being vendored directly into k8s anymore with the removal of log sanitization.

Comment (tabbysable) [reply vendor/modules.txt:1940]: Keeping the data policy tags in, so they could be used to benefit static analysis, was the primary concern raised during the SIG Security meeting. Thank you for calling this out explicitly.

**Comment by dashpole** [comment]
cc @vantuvt 
/triage accepted

**Comment by ehashman** [comment]
/assign @liggitt 

for top-level approval/final API pass

**Comment by liggitt** [comment]
/approve
config API change lgtm

**Pull request merged**